test/examples/uct_hello_world.c

```diff
@@ -1,17 +1,566 @@
 /**
-* Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
-*
+* Copyright (C) UT-Battelle, LLC. 2015. ALL RIGHTS RESERVED.
 * See file LICENSE for terms.
 */
 
-#define HAVE_CONFIG_H /* Force using config.h, so test would fail if header
-                         actually tries to use it */
+#include <ucs/type/status.h>
+#include <ucs/async/async.h>
 #include <uct/api/uct.h>
+#include <unistd.h>
 #include <stdio.h>
+#include <stdlib.h>
 
+#include "ucx_hello_world.h"
 
-int main(int argc, char **argv)
+typedef enum {
+    FUNC_AM_SHORT,
+    FUNC_AM_BCOPY,
+    FUNC_AM_ZCOPY
+} func_am_t;
+
+char * func_am_t2str(func_am_t func_am_type)
+{
+    switch (func_am_type) {
+    case FUNC_AM_SHORT: return "uct_ep_am_short";
+    case FUNC_AM_BCOPY: return "uct_ep_am_bcopy";
+    case FUNC_AM_ZCOPY: return "uct_ep_am_zcopy";
+    }
+    return NULL;
+}
+
+#define CHKERR_JUMP(cond, msg, label)       \
+do {                                        \
+    if (cond) {                             \
+    fprintf(stderr, "Failed to %s\n", msg); \
+    goto label;                             \
+    }                                       \
+} while (0)
+
+static void* desc_holder = 0;
+
+struct iface_info {
+    /* Interface attributes: capabilities and limitations */
+    uct_iface_attr_t attr;
+
```

Comment:
It was moved as is from exctive_message.c. If it was wrong, not a problem, I'll fix it